traceroute6.c

```diff
@@ -535,6 +536,13 @@ static int packet_ok(struct run_state *ctl, int cc, struct sockaddr_in6 *from,
 			nexthdr = *(unsigned char *)up;
 			up++;
 		}
+		if (nexthdr == 43) {
```

Comment:
These headers can occur in any order, so this whole skip process needs to be a loop.